python/sparkdl/estimators/keras_image_file_estimator.py

```diff
@@ -0,0 +1,284 @@
+        :return: list of MLlib models
+        """
+        transformers = []
+        for (param_map, model_bytes) in kerasModelsBytesRDD.collect():
+            model_filename = kmutil.bytes_to_h5file(model_bytes)
+            transformers.append({
+                'paramMap': param_map,
+                'transformer': KerasImageFileTransformer(modelFile=model_filename)})
+
+        return transformers
+
+    def _fitInParallel(self, dataset, paramMaps):
+        """
+        Fits len(paramMaps) models in parallel, one in each Spark task.
+        :param paramMaps: non-empty list or tuple of ParamMaps (dict values)
+        :return: list of fitted models, matching the order of paramMaps
+        """
+        sc = JVMAPI._curr_sc()
+        paramMapsRDD = sc.parallelize(paramMaps, numSlices=len(paramMaps))
+
+        # Extract image URI from provided dataset and create features as numpy arrays
+        localFeatures, localLabels = self._getNumpyFeaturesAndLabels(dataset)
+        localFeaturesBc = sc.broadcast(localFeatures)
+        localLabelsBc = None if localLabels is None else sc.broadcast(localLabels)
+
+        # Broadcast Keras model (HDF5) file content as bytes
+        modelBytes = self._loadModelAsBytes()
+        modelBytesBc = sc.broadcast(modelBytes)
+
+        # Obtain params for this estimator instance
+        baseParamMap = self.extractParamMap()
+        baseParamDict = dict([(param.name, val) for param, val in baseParamMap.items()])
+        baseParamDictBc = sc.broadcast(baseParamDict)
+
+        # Create useful parameters from paramMap
+        def _local_fit(override_param_map):
+            """
+            Fit locally a model with a combination of this estimator's param plus
+            with additional parameters provided by the input.
+            :param override_param_map: dict, key type is MLllib Param
+                                       They are meant to override the base estimator's params.
+            :return: serialized Keras model bytes
+            """
+            # Update params
+            params = baseParamDictBc.value
+            override_param_dict = dict([
+                (param.name, val) for param, val in override_param_map.items()])
+            params.update(override_param_dict)
+
+            # Create Keras model
+            model = kmutil.bytes_to_model(modelBytesBc.value)
+            model.compile(optimizer=params['optimizer'], loss=params['loss'])
+
+            # Retrieve features and labels and fit Keras model
+            features = localFeaturesBc.value
+            labels = None if localLabelsBc is None else localLabelsBc.value
+            _fit_params = params['kerasFitParams']
+            model.fit(x=features, y=labels, **_fit_params)
+
+            return kmutil.model_to_bytes(model)
+
+        kerasModelBytesRDD = paramMapsRDD.map(lambda paramMap: (paramMap, _local_fit(paramMap)))
+        return self._collectModels(kerasModelBytesRDD)
+
+    def _loadModelAsBytes(self):
```

Comment:
we will need to change `HasKerasModel` eventually: storing the name of a file is not enough for import/export. No need to change in this PR though.